src/sentry/snuba/metrics/extraction.py

```diff
@@ -304,11 +310,17 @@ def __init__(self, field: str, query: str):
 
         """
 
-        # On-demand metrics are implicitly transaction metrics. Remove the
-        # filter from the query since it can't be translated to a RuleCondition.
-        self._query = re.sub(r"event\.type:transaction\s*", "", query)
+        self._init__query(query)
         self._init_aggregate(field)
 
+    def _init__query(self, query: str) -> None:
+        # On-demand metrics are implicitly transaction metrics. Remove the
+        # filters from the query that can't be translated to a RuleCondition.
+        query = re.sub(r"event\.type:transaction\s*", "", query)
+        query = re.sub(r"project:\w+\s*", "", query)
```

Comment:
Can the project be quoted?

Reply:
Yes, handled with [2ea5a82](https://github.com/getsentry/sentry/pull/53053/commits/2ea5a820fbfeab0a3558c545660b00d35e02e608)